docs/atomic-swaps.md

```diff
@@ -0,0 +1,83 @@
+
+- Delegated secret management – The Solver is responsible for managing secrets, reducing operational complexity for users.  
+- Multiple Solver selection – Users can designate multiple Solvers to fulfill the transaction, mitigating Solver liveness risks.  
+- Incentive alignment – A reward/slash mechanism ensures that Solvers are economically incentivized to execute transactions promptly.  
+
+This document does not exhaustively cover all [edge cases](https://docs.train.tech/protocol-spec/edge-cases) and [implementation details](https://docs.train.tech) but instead provides a high-level overview of the core process.  
+
+## Protocol Walkthrough
+
+```mermaid
+sequenceDiagram
+    participant P1 as User
+    participant SC as Source Chain
+    participant P2 as Solver
+    participant DC as Destination Chain
+
+
+    P1->>SC: commit()
+    SC-->>P2: TokenCommitted
+    P2->>DC: lock()
+    DC-->>P1: TokenLocked(hashlock)
+    P1->>SC: addLock(hashlock)
+    SC-->>P2: TokenLockAdded
+    P2->>SC: redeem(secret)
+    SC-->>P1: User Funds Recieved 
+    P2->>DC: redeem(secret)
+    DC-->>P2: Solver Funds Recieved
+```
+
+The following outlines the end-to-end execution when Alice transfers 1 ETH from Starknet to Optimism.  
+
+### 1. Transaction initiation
+
+- Alice accesses the Bridge dApp, connects her wallet, and selects the Starknet → Optimism transfer route.  
+- The dApp queries the Solver Discovery contract to retrieve a list of available Solvers supporting the route.  
+- Alice submits a commit transaction on Starknet, locking 1 ETH and specifying the set of Solvers.  
+
+At this point, the locked funds are subject to one of two conditions:  
+
+- A `Hashlock` is added once the auction winner is determined.  
+- If no Solver successfully acts on this transaction before the `Timelock` expires, Alice can reclaim her funds.  
+
+### 2. Solver selection via Dutch Auction
+
+- A Dutch auction is initiated, where Solvers (e.g., Bob and John) compete by progressively lowering their bid price.  
+- Bob wins the auction, generates a secret `S`, computes `Hashlock = HASH(S)`, and locks 1 ETH for Alice on Optimism.  
+
+### 3. Local verification of the destination transaction
+
+- The dApp performs local verification by monitoring Bob’s transaction on Optimism.  
+- If a light client is available, it is utilized for validation. Otherwise, the dApp queries multiple RPC endpoints or a user-specified node.
+- Upon confirming the transaction, the `Hashlock` is retrieved.  
+
+### 4. Finalization
+
+- Once `Hashlock` is verified, Alice signs a transaction that assigns the `Hashlock` to her previously committed funds and sets the `receiver` to Bob.  
+
+At this stage, both chains (Starknet and Optimism) hold funds locked under the same `Hashlock`. The only remaining step is secret revelation.  
+
+### 5. Secret disclosure and fund release
+
+- Bob monitors Starknet for conf irmation that the commitment is locked in his favor with the expected `Hashlock`.  
```

Comment:
```suggestion
- Bob monitors Starknet for confirmation that the commitment is locked in his favor with the expected `Hashlock`.  
```
beware this tiny typo on the last commit :)

Reply:
Fixed.